packages/utilities/src/classNamesFunction.test.ts

```diff
@@ -0,0 +1,45 @@
+import { classNamesFunction } from './classNamesFunction';
+import { Stylesheet, IStyle } from '@uifabric/merge-styles';
+
+describe('classNamesFunction', () => {
+  let lastRule: string;
+
+  beforeEach(() => {
+    Stylesheet.getInstance().setConfig({
+      onInsertRule: () => {
+        /*no-op*/
+      }
+    });
+  });
+
+  afterEach(() => {
+    Stylesheet.getInstance().setConfig({
+      onInsertRule: undefined
+    });
+  });
+
+  it('can cache rules', () => {
+    let styleFunctionCalled = false;
+    const getClassNames = classNamesFunction<{ a: number; b: string }, { root: IStyle }>();
+    const getStyles = (props: { a: number; b: string }) => {
+      styleFunctionCalled = true;
+      return {
+        root: { width: props.a }
+      };
+    };
+
+    const classNames = getClassNames(getStyles, { a: 1, b: 'test' });
+    expect(styleFunctionCalled).toEqual(true);
+    styleFunctionCalled = false;
```

Comment:
nit: Move this line one line below or line 41 one line above so that `styleFunctionCalled=false` is always either just after or just before a blank line.